python/deltalake/writer.py

```diff
@@ -322,6 +313,34 @@ def default(self, obj: Any) -> Any:
         return json.JSONEncoder.default(self, obj)
 
 
+def try_get_table_and_table_uri(
+    table_or_uri: Union[str, Path, DeltaTable],
+    storage_options: Optional[Dict[str, str]] = None,
+) -> Tuple[Optional[DeltaTable], str]:
+    """Parses `table_or_uri` and returns `table` & `table_uri`.
+
+    Raises a ValueError if `table_or_uri` is not of type `str`, `Path` or `DeltaTable`
```

Comment:
Should we maybe write this in numpy docstring format? https://numpydoc.readthedocs.io/en/latest/format.html#raises

Reply:
I am very sorry, but I said something wrong. We are not using the numpy format... we are using the reST format. not sure how raises are done there. @wjones127 - do you know? I could not find good docs for that.

Reply:
Ah yes, the format shown [here](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format) looks very similar to what is used in other places in the codebase and it also specifies a way to describe a raise. I can rewrite it in that format if it's the correct one to use

Reply:
Yes that's the format. Could you rewrite that?